tests/integration/models/application.spec.coffee

```diff
@@ -226,6 +232,28 @@ describe 'Application Model', ->
 				promise = resin.models.application.generateProvisioningKey(999999)
 				m.chai.expect(promise).to.be.rejectedWith('Application not found: 999999')
 
+		describe 'resin.models.application.tags', ->
+
+			tagTestOptions =
+				model: resin.models.application.tags
+				resourceName: 'application'
+				uniquePropertyName: 'app_name'
+
+			beforeEach ->
+				tagTestOptions.resource = @application
```

Comment:
It's a bit clearer if you combine this with the above, rather than building the two halves separately. You lose the scoped variable, but mocha gives you a `this` context for exactly that:

```coffeescript
beforeEach ->
     @tagTestOptions = { ... }

describe '...' ->
    itShouldGetAllTags(@tagTestOptions)
```

Reply:
@pimterry This would error, since I'm using some of the properties to construct sensible titles during the `it()` calls, so calls like `it "should retrieve the tag by #{opts.resourceName} id"` will fail.
I initially tried to define a getter property/method on `tagTestOptions` that would reach `@application`, but obviously didn't manage to get the `this` scope of the `beforeEach` inside the `describe` call.

Reply:
That completely makes sense :+1: